websites/A/Animes Portal/presence.ts

```diff
@@ -0,0 +1,294 @@
+          .querySelector<HTMLDivElement>(
+            "body > main.animated > div.wrapper > article.rowView > aside.aside > div.cover-holder > img.abs"
+          )
+          ?.getAttribute("src");
+
+      if (image) {
+        presenceData.smallImageKey = "logo";
+        presenceData.smallImageText = "logo";
+      }
+
+      presenceData.details = `Viewing ${name}`;
+      presenceData.largeImageKey = image ?? "logo";
+      presenceData.buttons = [
+        {
+          label: "View",
+          url: document.location.href
+        }
+      ];
+    } else if (uid && eid) {
+      const { name, episode, part } = getInfo(),
+        playing = document
+          .querySelector(
+            "body > main.animated > div.wrapper > section.holder > div.player > div.holder > div.vplayer"
+          )
+          ?.classList.contains("playing");
+
+      if (!playing) {
+        presenceData.details = `Watching ${name}`;
+        presenceData.state = `Episode ${episode}`;
+        presenceData.buttons = [
+          {
+            label: "Watch",
+            url: document.location.href
+          }
+        ];
+
+        if (part) presenceData.state = presenceData.state += ` (part ${part})`;
+      } else return;
+    }
+  } else if (pathname.startsWith("/movies")) {
+    if (paths[1] === "search" && paths[2])
+      presenceData.details = `Searching for ${paths[2].replaceAll("-", " ")}`;
+    else if (paths[1])
+      presenceData.details = `Viewing movies that starts with the keyboard ${paths[1].toUpperCase()}`;
+    else presenceData.details = "Viewing movies";
+  } else if (pathname.startsWith("/movie/")) {
+    const name = document.querySelector(
+        "body > main.animated > div.wrapper > article.rowView > header.rowView-head > h1.heading"
+      ).textContent,
+      image = document
+        .querySelector(
+          "body > main.animated > div.wrapper > article.rowView > aside.aside > div.cover-holder > img.abs"
+        )
+        ?.getAttribute("src");
+
+    if (image) {
+      presenceData.smallImageKey = "logo";
+      presenceData.smallImageText = name;
+    }
+
+    presenceData.details = `Viewing movie ${name}`;
+    presenceData.largeImageKey = image ?? "logo";
+  } else if (pathname.startsWith("/manga")) {
+    if (paths[1] === "search" && paths[2])
+      presenceData.details = `Searching for ${paths[2].replaceAll("-", " ")}`;
+    else if (paths[1] && paths[2]?.startsWith("vol-")) {
+      const tom = paths[2].replace("vol-", "");
+
+      if (paths[3].startsWith("chapter-")) {
+        const list = document.querySelectorAll(
+            "body > main.animated > ul#path > li > a > span"
+          ),
+          name = list[1].textContent,
+          page = document.querySelector(
+            "body > main.animated > div.wrapper > div.heading > b#num"
+          ).textContent;
+
+        presenceData.details = `Reading manga ${name}`;
+        presenceData.state = `Tom: ${tom}, Chapter: ${between(
```

Comment:
What's tom?

Reply:
its the volume, like the arc
i will rename it to `Volume: ${tom}` or arc, idk